CHANGELOG.md

```diff
@@ -10,12 +10,289 @@ New features:
 
 Bugfixes:
 
-* Only include direct dependencies in the output for `purs graph` (#3993, @colinwahl)
+Other improvements:
 
-Fixes a bug where the transitive closure of a module's dependencies
-where included in the `depends` field in the output for `purs graph`.
+## v0.14.0
 
-Other improvements:
+### Polykinds
+
+Polymorphic kinds, based on the [Kind Inference for Datatypes](https://richarde.dev/papers/2020/kind-inference/kind-inference.pdf) paper (#3779, #3831, #3929, @natefaubion)
+
+Kinds are to types what types are to terms (although kinds are themselves types now), but whereas we have polymorphic types, kinds were monomorphic.
+
+This meant that we were not able to abstract over kinds, leading for instance to a proliferation of proxy types:
+
+```purescript
+data Proxy (a :: Type) = Proxy
+data SProxy (a :: Symbol) = SProxy
+data RProxy (row :: # Type) = RProxy
+data RLProxy (row :: RowList) = RLProxy
+```
+
+Now we can have a single proxy type, whose parameter has a polymorphic kind.
+
+#### Type In Type
+
+The type-checker now supports `TypeInType` (or `Type :: Type`), so the old `Kind` data type and namespace is now gone. Kinds and types are the same and exist in the same namespace.
```

Comment:
I wasn't personally familiar with `TypeInType`, which is -- as far as I know -- referring to the Haskell extension `-XTypeInType`. The [GHC documentation for the extension](https://downloads.haskell.org/~ghc/8.0.2/docs/html/users_guide/glasgow_exts.html#ghc-flag--XTypeInType) has a nice summary:

> Allow kinds to be as intricate as types, allowing explicit quantification over kind variables, higher-rank kinds, and the use of type synonyms and families in kinds, among other features.

Is this also true of our implementation (@natefaubion)? If so, is it worth adapting this sentence to help briefly note what the benefit of adding this feature was?

Reply:
I think this is all true, but `TypeInType` has become an alias for other extensions in 8.6.1 and was deprecated in 8.8.1, so perhaps it would be better not to mention it there? What would you think of renaming the title of this section to `Type :: Type` and drop the start of this sentence?

```suggestion
The old `Kind` data type and namespace is now gone. Kinds and types are the same and exist in the same namespace.
```

Reply:
For what it's worth, this change sounds reasonable to me.

Reply:
Sounds good to me.